### PR TITLE
Update testsuite

### PR DESCRIPTION
**Pull request by binji**
Includes the three merged proposals (nontrapping-float-to-int,
sign-extension, multi-value).

It also has one bug fix when parsing table limits (when the min or max
size does not fit in a 32-bit int).

### CODE REVIEW

File: test/wasm2c/spec/i32.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ;;; TOOL: run-spec-wasm2c
 ;;; STDIN_FILE: third_party/testsuite/i32.wast
+;;; SKIP: sign-extend, multi-value not supported yet
```
Comment (sbc100) [inline test/wasm2c/spec/i32.txt:3]: Does wabt not support these?

Comment (sbc100) [reply test/wasm2c/spec/i32.txt:3]: I don't get how these are marked as SKIP but still seem to report passing ?

Comment (binji) [reply test/wasm2c/spec/i32.txt:3]: wasm2c doesn't support this yet. And `SKIP` doesn't run the test, but it also doesn't check the output. I guess that's a bit confusing, but it means the diffs here are smaller.

Comment (sbc100) [reply test/wasm2c/spec/i32.txt:3]: Sorry I missed the sub directory name there.

**Pull request merged**